operator/objectsusercontroller/pipeline.go

```diff
@@ -45,9 +45,35 @@ func hasSecretRef(ctx *pipelineContext) bool {
 }
 
 func toConnectionDetails(csUser *cloudscalesdk.ObjectsUser) managed.ConnectionDetails {
+	if csUser == nil {
+		return map[string][]byte{}
+	}
+
+	if csUser.Keys == nil {
+		return map[string][]byte{}
+	}
+
+	if len(csUser.Keys) == 0 {
+		return map[string][]byte{}
+	}
+
+	if csUser.Keys[0] == nil {
+		return map[string][]byte{}
+	}
+
+	accessKey, exists := csUser.Keys[0]["access_key"]
+	if !exists {
+		accessKey = ""
+	}
+
+	accessSecret, exists := csUser.Keys[0]["secret_key"]
+	if !exists {
+		accessSecret = ""
+	}
+
```

Comment:
I don't like this solution, it's ok for avoid nil pointers, but root cause remains the same. If there is a problem with user in OBSERVE, we should trigger UPDATE. With this solution we might end up with working provider, but empty strings in conenction secret

Reply:
The nilpointer didn't happen in observe, it actually happened during the create.

I've checked the connection strings after deploying this for the affected objects, they were still populated as expected.

Reply:
Ok thanks for an update, I think this thread can be treated as resolved :)